sql/distsql: preserve JobID in redacted logs

spilchen opened the pull request:
Previously, the JobID log tag could be redacted, making it difficult to trace TTL job execution when viewing redacted logs. This change ensures the JobID is logged as a redact.SafeString, allowing it to appear in redacted outputs for easier debugging.

Additionally, the flowCtx.ID is now also saved as a safe string too.

Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146362)
<!-- Reviewable:end -->

rafiss reviewed (approved):
nice work, lgtm! since this is just logging/redaction related, and it will assist with supportability, let's backport this.

please get a Queries team member to sign off as well.

yuzefovich reviewed (commented):
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146362)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/sql/distsql/server.go` line 387 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/146362#-OPklRCu7tOGddGT1ZfW:-OPklRCv6cXA91ystEwi:bcfnjp4) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b59f402c6e41347efe2c54364a662bd6986e0fe/pkg/sql/distsql/server.go#L387)):*
> ```Go
> 	if !f.IsLocal() {
> 		bld := logtags.BuildBuffer()
> 		bld.Add("f", flowCtx.ID.Short())
> ```

Hm, I think we might need to do a similar change in more places if we want to remove `String` method call. `String` method produces hexadecimal representation, and it's what we use for the flow memory monitor name (see `mon.Name.SafeFormat`) which then shows up in `crdb_internal.node_memory_monitors`.

I think we should do
```
redact.SafeString(flowCtx.ID.Short().String())
```
for now. Feel free to leave `TODO(yuzefovich): consider using int32 representation instead of hexadecimal one.`


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146362)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @yuzefovich)

___
*[`pkg/sql/distsql/server.go` line 387 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/146362#-OPklRCu7tOGddGT1ZfW:-OPl3QQG2RXq2ymVBw2d:bukbgkd) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b59f402c6e41347efe2c54364a662bd6986e0fe/pkg/sql/distsql/server.go#L387)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

Hm, I think we might need to do a similar change in more places if we want to remove `String` method call. `String` method produces hexadecimal representation, and it's what we use for the flow memory monitor name (see `mon.Name.SafeFormat`) which then shows up in `crdb_internal.node_memory_monitors`.

I think we should do
```
redact.SafeString(flowCtx.ID.Short().String())
```
for now. Feel free to leave `TODO(yuzefovich): consider using int32 representation instead of hexadecimal one.`
</blockquote></details>

Good point. I added in your suggestion.


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
Thanks!

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146362#-:-OPl6eeL4CHmGJDsz1u1:b-ucfknh)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)



<!-- Sent from Reviewable.io -->

rafiss reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146362)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/sql/distsql/server.go` line 387 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/146362#-OPklRCu7tOGddGT1ZfW:-OPl7Xv3D2a18b6x3dp1:b-vef2fb) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b59f402c6e41347efe2c54364a662bd6986e0fe/pkg/sql/distsql/server.go#L387)):*
<details><summary><i>Previously, spilchen wrote…</i></summary><blockquote>

Good point. I added in your suggestion.
</blockquote></details>

are you sure there's a problem here?

```
func TestShortString(t *testing.T) {
	u := MakeV4()

	fmt.Printf("UUID: %s\nredacted UUID: %s\nuuid.String(): %s\n", u, redact.Sprint(u), u.String())
	s := u.Short()
	fmt.Printf("Short UUID: %s\nredacted short UUID: %s\nshort.String(): %s\n", s, redact.Sprint(s), s.String())
}
```

output:
```
UUID: 3a10d779-6336-4244-bfd5-076f51369cec
redacted UUID: 3a10d779-6336-4244-bfd5-076f51369cec
uuid.String(): 3a10d779-6336-4244-bfd5-076f51369cec
Short UUID: 3a10d779
redacted short UUID: 3a10d779
short.String(): 3a10d779
```


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146362)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/sql/distsql/server.go` line 387 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/146362#-OPklRCu7tOGddGT1ZfW:-OPl8xFMC611S0QnGd0J:bbt3ot6) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b59f402c6e41347efe2c54364a662bd6986e0fe/pkg/sql/distsql/server.go#L387)):*
<details><summary><i>Previously, rafiss (Rafi Shamim) wrote…</i></summary><blockquote>

are you sure there's a problem here?

```
func TestShortString(t *testing.T) {
	u := MakeV4()

	fmt.Printf("UUID: %s\nredacted UUID: %s\nuuid.String(): %s\n", u, redact.Sprint(u), u.String())
	s := u.Short()
	fmt.Printf("Short UUID: %s\nredacted short UUID: %s\nshort.String(): %s\n", s, redact.Sprint(s), s.String())
}
```

output:
```
UUID: 3a10d779-6336-4244-bfd5-076f51369cec
redacted UUID: 3a10d779-6336-4244-bfd5-076f51369cec
uuid.String(): 3a10d779-6336-4244-bfd5-076f51369cec
Short UUID: 3a10d779
redacted short UUID: 3a10d779
short.String(): 3a10d779
```
</blockquote></details>

Huh, ok, I was wrong. I assumed that because `uuid.Short` is marked as `redact.SafeValue`, we'd get some different behavior than `redact.SafeString(uuid.Short.String())`, but that assumption doesn't make sense - the same `String()` implementation will be called under the hood. Thanks for double-checking! The original version LGTM.


<!-- Sent from Reviewable.io -->

spilchen committed:
sql: preserve JobID in redacted logs

Previously, the JobID log tag could be redacted, making it difficult to
trace TTL job execution when viewing redacted logs. This change ensures
the JobID is logged as a redact.SafeString, allowing it to appear in
redacted outputs for easier debugging.

Additionally, the flowCtx.ID is now also saved as a safe string too.

Epic: none
Release note: none

spilchen commented:
TFTRs!

bors r+
